numpy/random/common.pyx

```diff
@@ -227,7 +227,7 @@ cdef check_output(object out, object dtype, object size):
         raise ValueError('Supplied output array is not contiguous, writable or aligned.')
     if out_array.dtype != dtype:
         raise TypeError('Supplied output array has the wrong type. '
-                        'Expected {0}, got {0}'.format(dtype, out_array.dtype))
+                        'Expected {0}, got {1}'.format(dtype, out_array.dtype))
```

Comment:
Thanks. It probably makes sense to call `dtype = np.dtype(dtype)` for nicer parsing (that step probably happens anyway during the `!=` operation, even if it is a bit hidden. Or just call it for the printing.

Reply:
Added `np.dtype(dtype)` to the `format()` command in the latest commit. Example output now: `TypeError: Supplied output array has the wrong type. Expected float64, got float32`